### ISSUE TITLE
Crashing upon saving

### ISSUE THREAD

**Issue by xblackbytesx** [T-Thurow/lychee-slicer-flake#1]
Hi there :-)

Sorry to bother you with this but I stumbled upon this flake wrapper and was wondering if you've been able to get LycheeSlicer running successfully in NixOS using your flake. 

I've just about tried everything and the only thing that works out for me is to use the flathub version but that one is stuck at 6.x.x for a while now and besides that I'd rather use my home-manger/nixos config to install all packages.

I was very happy to see a fellow 3D print nerd working on this and immediately decided to try your wrapper out.  Sadly while it does install correctly it suffers the same issue as all other methods I've tried and that is that you simply can't save a `.lys` file without it immediately crashing.

I've seem pretty much every Linux user outside of the debian spinnoff  folks suffer from this issue and was really curious if you've managed to get it to work correctly.

In the spirit of sharing an contribution, here was my attempt at a wrapper that I used earlier. It also adds a "launcher" for your desktop manager of choice. Perhaps this is useful to add to your wrapper as well since I noticed  you didn't have any yet.

Really curious to hear from you :-) thanks a lot in advance!

```
{ lib, stdenv, pkgs, appimageTools, gsettings-desktop-schemas, gtk3, fetchurl, makeDesktopItem, ...}:

let
  version = "7.0.1";
  pname = "lychee-slicer";
  appImage = fetchurl {
    url = "https://mango-lychee.nyc3.cdn.digitaloceanspaces.com/LycheeSlicer-${version}.AppImage";
    sha256 = "sha256-p55PUT1z4mDCc5e7ZY+Ty1jffbkBcJ1e8OMn5s7Y7hw=";
  };
  desktopItem = makeDesktopItem {
    name = pname;
    exec = pname;
    icon = pname;
    desktopName = "Lychee Slicer";
    genericName = "3D Slicer";
    comment = "Prepare 3D models for SLA 3D printing";
    categories = [ "Graphics" "3DGraphics" ];
  };
in
appimageTools.wrapType2 {
  inherit pname version;
  src = appImage;

  profile = ''
    export LC_ALL=C.UTF-8
    export XDG_DATA_DIRS="${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}:${gtk3}/share/gsettings-schemas/${gtk3.name}:$XDG_DATA_DIRS"
  '';

  extraPkgs = pkgs: with pkgs; [
    gsettings-desktop-schemas
    glib
    gtk3
    gnome.adwaita-icon-theme
  ] ++ (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs);

  extraInstallCommands = ''
    mkdir -p $out/share/applications
    mkdir -p $out/share/icons/hicolor/256x256/apps
    cp ${desktopItem}/share/applications/${pname}.desktop $out/share/applications/
    ln -s $out/bin/${pname} $out/share/icons/hicolor/256x256/apps/${pname}
  '';
}
```

**Comment by T-Thurow** [comment]
Hi @xblackbytesx thank you for sharing your wrapper attempt. Adding the software to the launcher would certainly be a good improvement. :+1: 

I just tried saving and I'm running in the same issue as you. It freezes and then crashes with a window asking if i want to restart the program. Same for the autosave feature. Funny enough everything else like loading, slicing and saving the sliced object works. 

My Arch Linux Machine is also impacted by the same issue (using the deb package wrapper from AUR). 
And Debian based platforms like Ubuntu don't have that issue?

**Comment by T-Thurow** [comment]
Okay i tested with Ubuntu 24.04.1 LTS and installed the deb package. Same issue. There seems to be something wrong with Lychee itself.

**Comment by xblackbytesx** [comment]
Hi @T-Thurow  :-) thanks for taking the time to check on your side as well!

Yeah my Arch used to suffer the same issue as well. The entire Linux 3D print community already complained to Mango3D for them to look into this issue yet they refuse to do anything about it. I've personally complained about this in their Discord and provided some logs but to no avail.

It seems they like selling us the software yet want nothing to do with Linux after the sale. Pretty bad stuff. I cancelled my subscription with them because of this issue. The only way to successfully run it without crashes seems to be to use the (pretty outdated) Flatpak build on Flathub.

I've given up hope that Mango will fix things from their end so now it's time to work around them I guess. Either a VM or to see if a WINE based approach would work. Far from ideal though. Thanks for your replies and as soon as I find a proper method to run the software without the crashes I'll drop you a line here.

Have a good one!

**Comment by T-Thurow** [comment]
@xblackbytesx this reaction of Mango3d is disappointing to the say the least.
But your flathub approach gave me an idea.
I tried rebuilding the flatpak with Lychee 7.1.1 (skipped 7.1.2 since it introduces bugs. At least the people on discord complain about that version).
I got 7.1.1 to work with the rebuilded flatpak without the save file bug.
I will commit a pull request to the flatpak repo tomorrow.

**Comment by xblackbytesx** [comment]
@T-Thurow Haha yeah I had the same plan a few weeks ago to do exactly that. However I noticed there already was a PR open to upgrade to the 7.x.x tag but it seems nobody is actively merging those in and/or looking at the PR's. But I am curious how you managed to build and install the Flatpak locally. If you don't mind sharing that I'd love to have that as a workaround as well.

**Comment by T-Thurow** [comment]
@xblackbytesx The process is explained in the nix [FlatPak](https://nixos.wiki/wiki/Flatpak) documentation under development.
Aside from that i had to install ```appstream``` for building.  I just used it in a shell with ```nix-shell -p appstream```.
Then to build and install locally i used 
```
flatpak-builder --force-clean --user --install-deps-from=flathub --install _flatpak io.mango3d.LycheeSlicer.yml
```
inside the cloned repo.
By updating the url, sha256 and size in ```io.mango3d.LycheeSlicer.yml``` you can set which Lychee version should be used for building.

**Comment by xblackbytesx** [comment]
Ah sorry for not doing that research myself but thanks a ton for your explanation! I never dabbled into packaging my own Flatpak before so that's definitely something I'll be trying out now. For now I noticed you opened your PR over at the Flathub repo and managed to install your test build by running (for others that stumble upon our topic): 

```
flatpak install --user https://dl.flathub.org/build-repo/149102/io.mango3d.LycheeSlicer.flatpakref
```
Gotta say this build seems to work great so far! This will at least get me slicing reliably again for the time being. Ideally Mango would get their act together and actually support Linux properly but this is a nice workaround.

Thanks for this awesome back and forward and your contribution :+1: 

Happy Holidays (soon-ish)!

EDIT: Fixed typo
